### PR TITLE
Add support for Azure AAD login. Fix some bugs.

### PR DESCRIPTION
**Pull request by brendandburns**
@drubin

### CODE REVIEW

**Comment by brendandburns** [comment]
@drubin fixed lint issues.

File: kubernetes/src/kubernetes/config/kube_config.rb [hunk 1]
```diff
@@ -120,15 +122,20 @@ def load_token_file(user)
...
 
+    # rubocop:disable AbcSize
     def setup_auth(user)
       # Convert token field to http header
       if user['token']
         user['authorization'] = "Bearer #{user['token']}"
       elsif user['username'] && user['password']
         user_pass = "#{user['username']}:#{user['password']}"
         user['authorization'] = "Basic #{Base64.strict_encode64(user_pass)}"
+      elsif user['auth-provider'] && user['auth-provider']['name'] == 'azure'
```
Comment (drubin) [inline kubernetes/src/kubernetes/config/kube_config.rb]: Should we not add a test? :)

Comment (brendandburns) [reply kubernetes/src/kubernetes/config/kube_config.rb]: done.

**Comment by brendandburns** [comment]
@drubin finally got around to adding a test.

File: kubernetes/spec/fixtures/config/kube_config_hash.rb [hunk 2]
```diff
@@ -167,6 +180,18 @@
     TEST_USER_SIMPLE_TOKEN_FILE,
     TEST_USER_USER_PASS,
     TEST_USER_CERT_DATA,
+<<<<<<< HEAD
```
Comment (drubin) [inline kubernetes/spec/fixtures/config/kube_config_hash.rb]: @brendandburns you have some merge conflicts here.

**Comment by brendandburns** [comment]
@drubin merge errors fixed. (apologies!)

**Comment by drubin** [comment]
/lgtm

No worries.

**Pull request merged**